src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -4,57 +4,103 @@ package freechips.rocketchip.diplomacy
+  protected[diplomacy] val parent: Option[LazyModule] = LazyModule.scope
 
-  // code snippets from 'InModuleBody' injection
+  /** code snippets from [[InModuleBody]] injection. */
   protected[diplomacy] var inModuleBody = List[() => Unit]()
 
+  /** chain of parents, [[parent]] on left, top on right. */
   def parents: Seq[LazyModule] = parent match {
     case None => Nil
     case Some(x) => x +: x.parents
   }
 
+  /** [[LazyModule.scope]] stack push. */
   LazyModule.scope = Some(this)
+  /** inject this instance to [[parent]] if this is not top. */
   parent.foreach(p => p.children = this :: p.children)
 
-  // suggestedName accumulates Some(names), taking the final one. Nones are ignored.
+  /** accumulates Some(names), taking the final one. Nones are ignored. */
   private var suggestedNameVar: Option[String] = None
+
+  /** override Verilog module name of [[LazyModuleImpLike]] module.
+    * @param x suggested name to [[LazyModuleImpLike]] module.
+    * */
   def suggestName(x: String): this.type = suggestName(Some(x))
   def suggestName(x: Option[String]): this.type = {
     x.foreach { n => suggestedNameVar = Some(n) }
     this
   }
 
+  /** helper of [[className]]. */
   private def findClassName(c: Class[_]): String = {
     val n = c.getName.split('.').last
     if (n.contains('$')) findClassName(c.getSuperclass) else n
   }
 
-  lazy val className = findClassName(getClass)
-  lazy val suggestedName = suggestedNameVar.getOrElse(className)
-  lazy val desiredName = className // + hashcode?
+  /** get the class name of this instance. */
+  lazy val className: String = findClassName(getClass)
+  /** used by [[LazyModuleImpLike.suggestName]]. */
+  lazy val suggestedName: String = suggestedNameVar.getOrElse(className)
+  /** used by [[LazyModuleImpLike.desiredName]]. */
+  lazy val desiredName: String = className // + hashcode?
 
+  /** @return instance name. */
   def name = suggestedName // className + suggestedName ++ hashcode ?
+  /** @return line to define this instance. */
   def line = sourceLine(info)
 
-  // Accessing these names can only be done after circuit elaboration!
-  lazy val moduleName = module.name // The final Verilog Module name
+  /** Accessing these names can only be done after circuit elaboration! */
+  /** The final Verilog Module name. */
+  lazy val moduleName = module.name
+  /** The final Verilog Module name with hierarchy. */
   lazy val pathName = module.pathName
-  lazy val instanceName = pathName.split('.').last // The final Verilog instance name
+  /** The final Verilog instance name. */
+  lazy val instanceName = pathName.split('.').last
 
+  /** chisel hardware implementation of this [[LazyModule]],
+    * This module should be define by `lazy val` for lazy evaluation.
+    * Generally, it is the beginning of phase 2.
+    * */
   def module: LazyModuleImpLike
 
-  def omitGraphML: Boolean = !nodes.exists(!_.omitGraphML) && !children.exists(!_.omitGraphML)
+  /** generate the [[graphML]]. */
```

Comment:
```suggestion
  /** generate the [[graphML]] representation for this instance. This is a representation of the Nodes, edges, Lazy Module hierarchy, and any other information that is added in by the implementations. It can be converted to a graphical representation of the [[LazyModule]] hierarchy with various third-party tools.*/
```